src/transactions/feed/queryHelper.ts

```diff
@@ -134,23 +142,36 @@ export function useFetchTransactions(): QueryHookResult {
     }
   )
 
+  useEffect(() => {
+    // this hook ensures that we populate the entire screen with transactions
+    // on load so that future refetches can be correctly triggered by
+    // `onEndReached`, in the event that blockchain-api returns a small number
+    // of results for the first page(s)
+    const { transactions, pageInfo } = fetchedResult
+    if (!loading && transactions.length < MIN_NUM_TRANSACTIONS && pageInfo?.hasNextPage) {
```

Comment:
When I try `if (!loading && transactions.length < MIN_NUM_TRANSACTIONS || pageInfo?.hasNextPage)` the feed fetches all the transactions and I am seeing more than if I scroll down with the current implementation 👀

Reply:
Fixed in https://github.com/valora-inc/wallet/pull/2779/commits/9294c9d01f30bad4b262db0b9f4b52e79a782958!

Reply:
are you sure? when i inspect the network, i see blockchain-api called until 10 transactions are received. it may look like there are many feed items because 10 is more than one page especially on smaller devices. we also have the `onEndReached` which triggers to fetch more transactions before you get to the bottom of the page.

i can verify that not all transactions are fetched - using the account i'm testing with, this would be a massive amount of requests to blockchain-api and i'm definitely not seeing this

Reply:
thanks for flagging though! i did double check